stdpopsim/slim_engine.py

```diff
@@ -704,6 +697,79 @@ def slim_array_string(iterable, indent, width=80):
     )
 
 
+def _simulate_mutation_type_in_slim(mt):
+    """
+    Returns true if this `stdpopsim.MutationType` is to be simulated by SLiM.
+    """
+    return not mt.is_neutral
+
+
+def _mutation_types_proportions_slim(d):
+    """
+    Returns a list with the slim proportions for each mutation type in this
+    `stdpopsim.DFE`.  A MutationType that is not to be simulated by SLiM will
+    have 0.0 proportion in SLiM, unless this DFE is completely neutral. Because
+    SLiM does not allow a GenomicElementType to have all it's MutationTypes to
+    have 0.0 proportion. This works because all the intervals in this DFE will
+    have 0.0 mutation rate anyway, as computed by
+    `get_msp_and_slim_mutation_rate_maps`.
+    """
+    return [
+        prop if _simulate_mutation_type_in_slim(mt) or d.is_neutral else 0.0
+        for prop, mt in zip(d.proportions, d.mutation_types)
+    ]
+
+
+def add_top_level_metadata(ts, contig):
```

Comment:
Wait, what's the idea here? We want SLiM itself to be storing this information, because then we don't have to coordinate between the python and SLiM code so much?